docs/tutorials/hpo-workflows/hpo-workflows.rst

```diff
@@ -0,0 +1,334 @@
+.. _hpo_workflows:
+
+Hyperparameter Optimization with RayTorchTrainer on MNIST
+=========================================================
+
+This tutorial provides a step-by-step guide to using the ``RayTorchTrainer`` class for running 
+a hyperparameter optimization (HPO) study. We assume that you are familiar with the ``TorchTrainer`` class, and the itwinai training pipeline. If you are not, you might want to go through the tutorials on these first.
+To illustrate the process, we will work with the FashionMNIST dataset.
+
+The code we write here will closely resemble what you would implement using the itwinai 
+``TorchTrainer``, with minor modifications required to make it compatible with the Ray framework. 
+By the end of this tutorial, you will:
+
+*   Understand how the ``RayTorchTrainer`` functions.
+*   Know how to create a configuration file to define your HPO study.
+*   Understand the steps required to define and run an HPO study with the itwinai training pipeline.
+
+You can find the full code for this tutorial `on Github <https://github.com/interTwin-eu/itwinai/blob/main/tutorials/hpo-workflows>`_.
+
+
+Setting up the Trainer
+-----------------------
+
+Let’s start by defining our trainer class. Its structure will closely mirror that of 
+the ``TorchTrainer``, but with a few key adjustments to integrate seamlessly with the Ray backend. 
+The most notable change is that we move certain initialization steps into the ``train()`` function. 
+Ray executes the ``train()`` function for each trial independently,
+which means that the trials are completetely agnostic of one another. This means that
+the distributed strategy and thus the model, optimizer, logger, etc., can only be instantiated
+within the ``train()`` function.
+For more details on this, see the :doc:`HPO introduction <../../how-it-works/hpo/explain-hpo>`.
+
+In this tutorial, we will tune two hyperparameters, the batch size and the learning rate.
+Our model will be a ResNet18, and we will train it on the FashionMNIST dataset.
+
+
+Code Comparison: RayTorchTrainer vs TorchTrainer
+------------------------------------------------------
+
+.. tabs::
+
+    .. tab:: RayTorchTrainer
+
+        .. code-block:: python
+
+            class MyRayTorchTrainer(RayTorchTrainer):
+                def __init__(
+                    self,
+                    config: Dict,
+                    strategy: Literal["ddp", "deepspeed"] = "ddp",
```

Comment:
I don't know if this is important or not, so I'll let you decide: If you wanted to copy this directly, you would have to import `Literal` from `typing`, so right now it is not possible to just copy paste without changes. Probably doesn't matter though, just something I noticed.

Reply:
Could you explain more what you mean? Copy and paste this where?

Reply:
Honestly, it probably doesn't matter. People would have to fix tons of imports anyway, so it's fine the way it is.